vnext/Directory.Build.props

```diff
@@ -16,8 +16,8 @@
     -->
     <EnableSourceLink Condition="'$(EnableSourceLink)' == ''">false</EnableSourceLink>
     <!-- When bumping the Folly version, be sure to bump the git hash of that version's commit and build Folly.vcxproj (to update its cgmanifest.json) too. -->
-    <FollyVersion>2023.10.16.00</FollyVersion>
-    <FollyCommitHash>3da4d3ccc4ea69bc10800754816e339d912736bd</FollyCommitHash>
+    <FollyVersion>2023.11.06.00</FollyVersion>
```

Comment:
This means most of https://github.com/microsoft/react-native-windows/tree/main/vnext/Folly/TEMP_UntilFollyUpdate can (and probably should be) deleted